src/Spice86.Core/Emulator/CPU/CPU.cs

```diff
@@ -85,30 +84,35 @@ public class Cpu : IDebuggableComponent {
 
     public InterruptVectorTable InterruptVectorTable { get; }
 
-    public Cpu(IMemory memory, State state, DualPic dualPic, IOPortDispatcher ioPortDispatcher, CallbackHandler callbackHandler, MachineBreakpoints machineBreakpoints, ILoggerService loggerService, ExecutionFlowRecorder executionFlowRecorder, bool recordData) {
+    public Cpu(InterruptVectorTable interruptVectorTable,
+        Stack stack, FunctionHandler functionHandler, FunctionHandler functionHandlerInExternalInterrupt,
+        IMemory memory, State state, DualPic dualPic,
+        IOPortDispatcher ioPortDispatcher, CallbackHandler callbackHandler, EmulatorBreakpointsManager emulatorBreakpointsManager,
+        ILoggerService loggerService, ExecutionFlowRecorder executionFlowRecorder) {
         _loggerService = loggerService;
         _memory = memory;
         State = state;
         DualPic = dualPic;
         _ioPortDispatcher = ioPortDispatcher;
         _callbackHandler = callbackHandler;
-        MachineBreakpoints = machineBreakpoints;
-        InterruptVectorTable = new(_memory);
-        _alu8 = new Alu8(state);
-        Stack = new Stack(_memory, state);
+        EmulatorBreakpointsManager = emulatorBreakpointsManager;
         ExecutionFlowRecorder = executionFlowRecorder;
-        FunctionHandler = new FunctionHandler(_memory, state, ExecutionFlowRecorder, _loggerService, recordData);
-        FunctionHandlerInExternalInterrupt = new FunctionHandler(_memory, state, ExecutionFlowRecorder, _loggerService, recordData);
+        InterruptVectorTable = interruptVectorTable;
+        _alu8 = new(state);
```

Comment:
If we can move the AAA,AAD,AAM,AAS,DAA and DAS instructions to one of the Instructions classes, then we don't need this _alu8 and we can make the construction of the Instruction classes the same.

Reply:
Done.